test/e2e/set/scyllacluster/datainserter.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2021 ScyllaDB
+
+package scyllacluster
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sort"
+
+	"github.com/gocql/gocql"
+	"github.com/scylladb/gocqlx/v2"
+	"github.com/scylladb/gocqlx/v2/table"
+	scyllav1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const nRows = 10
+
+type DataInserter struct {
+	client            corev1client.CoreV1Interface
+	session           *gocqlx.Session
+	table             *table.Table
+	data              []*test
+	replicationFactor int
+}
+
+type test struct {
+	Id   int
+	Data string
+}
+
+func NewDataInserter(ctx context.Context, client corev1client.CoreV1Interface, sc *scyllav1.ScyllaCluster, replicationFactor ...int) (*DataInserter, error) {
+	rf := 0
+	if len(replicationFactor) > 0 {
+		rf = replicationFactor[0]
+	} else {
+		// Default to a total number of members
+		for _, r := range sc.Spec.Datacenter.Racks {
+			rf += int(r.Members)
+		}
+	}
+
+	session, err := newSession(ctx, client, sc)
+	if err != nil {
+		return nil, err
+	}
+
+	d := &DataInserter{
+		client:  client,
+		session: session,
+		table: table.New(table.Metadata{
+			Name:    "test.test",
+			Columns: []string{"id", "data"},
+			PartKey: []string{"id"},
+		}),
+		data: func() []*test {
+			var lst []*test
+			for i := 0; i < nRows; i++ {
+				lst = append(lst, &test{Id: i, Data: rand.String(32)})
+			}
+			return lst
+		}(),
+		replicationFactor: rf,
+	}
+
+	return d, nil
+}
+
+func (d *DataInserter) Close() {
+	d.session.Close()
+}
+
+// UpdateClientEndpoints closes an existing session and opens a new one.
+// In case an error was returned, DataInserter can no Longer be used.
+func (d *DataInserter) UpdateClientEndpoints(ctx context.Context, sc *scyllav1.ScyllaCluster) error {
+	d.session.Close()
+
+	session, err := newSession(ctx, d.client, sc)
+	if err != nil {
+		return err
+	}
+	d.session = session
+
+	return nil
+}
+
+func (d *DataInserter) Write() error {
+	err := d.session.ExecStmt(`DROP KEYSPACE IF EXISTS test`)
```

Comment:
I think I'd split it into a separate function, `Clear` and `Insert`, but if we go for unique tables, `Clear` might not be needed.

Reply:
removed dropping keyspace, added IF NOT EXISTS to both keyspace and table creation. `Clear` is not necessary as keyspace is now created from UID